### PR TITLE
Add code to styleAs Text prop

### LINKED ISSUE #3124: Add code styleAs to Text.
### Area

UI Components

### The problem

Currently the only way to consume the --salt-typography-fontFamily-code style is via CSS token. It would be nice if we could consumer it via a component prop too.

### The solution

Add styleAs="code"

### Alternatives and examples

.

### Are you a JPMorgan Chase & Co. employee?

- [ ] I am an employee of JPMorgan Chase & Co.

### PR DESCRIPTION
**Pull request by libertymayc**
closes #3124

### CODE REVIEW

File: packages/core/src/text/TextCode.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { forwardRef } from "react";
+import { Text, TextProps } from "./Text";
+
+export const TextCode = forwardRef<
```
Comment (joshwooding) [inline packages/core/src/text/TextCode.tsx:4]: This will probably be better as "Code"

**Review by joshwooding — CHANGES_REQUESTED** [review]
This should be done the same way as headers so the styles are applied to `code.saltText` and `.saltText-code`

File: packages/core/src/text/Code.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { forwardRef } from "react";
+import { Text, TextProps } from "./Text";
+
+export const Code = forwardRef<HTMLSpanElement, Omit<TextProps<"span">, "as">>(
+  function TextAction({ children, className, ...rest }, ref) {
+    return (
+      <Text as="code" styleAs="code" ref={ref} {...rest}>
```
Comment (joshwooding) [inline packages/core/src/text/Code.tsx:7]:
```suggestion
      <Text as="code" ref={ref} {...rest}>
```

File: packages/core/src/text/Code.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { forwardRef } from "react";
+import { Text, TextProps } from "./Text";
+
+export const Code = forwardRef<HTMLSpanElement, Omit<TextProps<"span">, "as">>(
```
Comment (joshwooding) [inline packages/core/src/text/Code.tsx:4]:
```suggestion
export const Code = forwardRef<HTMLSpanElement, Omit<TextProps<"code">, "as">>(
```
Is it possible to use code here?

File: .changeset/silly-forks-nail.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+---
+
+Added "code" option to Text 'styleAs' prop
+
+```tsx
+<>
+  <Code>Code text</Code>
+  <Text styleAs="code">Text styled as code</Text>
+</>
+```
```
Comment (joshwooding) [inline .changeset/silly-forks-nail.md:12]: Realised we've had a changelog like this before so we should just copy it :)

```suggestion
`Text` has a new variant: `Code`. See [Text component](/salt/components/text) for more information.

```tsx
<>
  <Code>Code text</Code>
  <Text as="code">Text styled as code</Text>
  <Text styleAs="code">Text styled as code</Text>
</>
```
```

**Review by joshwooding — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Review by joshwooding — CHANGES_REQUESTED** [review]
Actually one small thing, can you add the code style to the site under the "Styles" example please

**Review by joshwooding — APPROVED** [review]

**Review by shey-v — APPROVED** [review]

**Pull request merged**